### ISSUE TITLE
Error when an identity isn't set should say that an identity isn't set

### ISSUE THREAD

**Issue by jakebeal** [SynBioDex/pySBOL3#177]
Here is a place where an inscrutable error could be made more scrutable:

```
import sbol3

doc = sbol3.Document()
sbol3.set_namespace('http://foo.org')
c = sbol3.Component('bar',sbol3.SBO_DNA)
doc.add(c)

lsc = sbol3.LocalSubComponent(sbol3.SBO_DNA)
# c.features += {lsc}  ## If you uncomment, last line will not error
int = sbol3.Interaction(sbol3.SBO_DEGRADATION)
p = sbol3.Participation(sbol3.SBO_REACTANT,lsc) # Error should explain the lsc is a child object that needs to be added before being used
```

**Comment by tcmitchell** [comment]
> 1st question: Is there a way to know all child classes in order to raise the error child object needs to be added before being used when the user forgets to do so?

Yes. Any class that is an instance of `Identified` and _not_ an instance of `TopLevel`.

> 2nd question: Should we raise that error inside the Participation class or the refobj_property class? I mean refobj_property will raise 'Object identity is uninitialized' before we mention that 'child object needs to be added before being used'.

Raising this error in the `Participation` class will only resolve this one specific case of a more general problem. I think the intent of the original message was to give one instance of a more general problem. I think what @jakebeal is after is a modification to the current error message that takes into account a class that probably needs to be added to a parent before being used. I think I'm saying that this logic would need to go in the `refobj_property`

**Comment by tcmitchell** [comment]
I have removed the "good first issue" label. This is not as simple as it would seem. The problem is that this is in the referenced object property, upon which `Identified` and `TopLevel` depend. So we've git a circularity issue. The referenced object property needs to know if the value is `Identified` and not `TopLevel`, and the `Identified` and `TopLevel` classes have not yet been defined when this file would get loaded. At least in my understanding.

We would need to resolve this with some Python tricks to determine inheritance without access to the `Identified` and `TopLevel` classes. With this in mind, I removed the "good first issue" tag.
